lxd/instance/drivers/driver_qemu.go

```diff
@@ -1571,6 +1579,58 @@ func (d *qemu) Start(stateful bool) error {
 	return nil
 }
 
+// advertiseVsockAddress advertises the CID and port to the VM.
+func (d *qemu) advertiseVsockAddress() error {
+	// Only advertise the vsock address if security.devlxd is true.
+	if !(d.expandedConfig["security.devlxd"] == "" || shared.IsTrue(d.expandedConfig["security.devlxd"])) {
+		return nil
+	}
+
+	client, err := d.getAgentClient()
+	if err != nil {
+		return fmt.Errorf("Failed getting agent client handle: %w", err)
+	}
+
+	agent, err := lxd.ConnectLXDHTTP(nil, client)
+	if err != nil {
+		return fmt.Errorf("Failed connecting to lxd-agent: %w", err)
+	}
+
+	defer agent.Disconnect()
+
+	req := struct {
```

Comment:
This isn't what I meant. While using a temporary struct is a little clearer here, what I really meant was to use a dedicated struct type in both the sender and the receiver so we can see where the communication is going to/from, and easily find where any field changes would break other users. In short it shows the relationship between sender and receiver in the code.